CONTRIBUTING.md

```diff
@@ -1,73 +1,143 @@
+4. You can clean up the dev setup by running `make dev-cleanup`.
 
-The following environment variables need to be set for acceptance tests to run:
+## Repository structure
 
-* `SNOWFLAKE_ACCOUNT` - The account name
-* `SNOWFLAKE_USER` - A snowflake user for running tests.
-* `SNOWFLAKE_PASSWORD` - Password for that user.
-* `SNOWFLAKE_ROLE` - Needs to be ACCOUNTADMIN or similar.
-* `SNOWFLAKE_REGION` - Default is us-west-2, set this if your snowflake account is in a different region.
-* `TF_ACC` - to enable acceptance tests.
+The notable technical files/directories inside the repository:
 
-For example:
+- `Makefile` - contains instructions to set up the developer's environment, run tests, etc.
+- `pkg/provider` - definition of the provider
+- `pkg/resources`, `pkg/datasources` - definitions and tests (consult section [Running the tests locally](#running-the-tests-locally) below) for resources and datasources
+- `pkg/acceptance` - helpers for acceptance and integration tests
+- `pkg/sdk` - definitions of the SDK objects (SDK is our client to Snowflake, using [gosnowflake driver](https://github.com/snowflakedb/gosnowflake) underneath)
+- `pkg/sdk/testint` - integration tests for the SDK (consult section [Running the tests locally](#running-the-tests-locally) below)
 
-```sh
-export SNOWFLAKE_ACCOUNT=TESTACCOUNT
-export SNOWFLAKE_USER=TEST_USER
-export SNOWFLAKE_PASSWORD=hunter2
-export SNOWFLAKE_ROLE=ACCOUNTADMIN
-export SNOWFLAKE_REGION=us-west-2
-export TF_ACC=true
-```
+**⚠️ Important ⚠️** We are in progress of cleaning up the repository structure, so beware of the changes in the packages/directories.
+
+## Running the tests locally
 
-You can also read the config from a `~/.snowflake/config` file, although you will still need to set `TF_ACC` to true.
+Currently, we have three main types of tests:
+- SDK unit tests (in directory `pkg/sdk`, files ending with `_test.go`)
+- SDK integration tests (in directory `pkg/sdk/testint`, files ending with `_integration_test.go`)
+- resource/datasource acceptance tests (in directories `pkg/resources` and `pkg/datasources`, files ending with `_acceptance_test.go`)
 
+Both integration and acceptance tests require the connection to Snowflake (some of the tests require multiple accounts).
+
+The preferred way of running particular tests locally is to create a config file `~/.snowflake/config`, with the following content.
 
-~/.snowflake/config
 ```sh
 [default]
-account='TESTACCOUNT'
-user='TEST_USER'
-password='hunter2'
-role='ACCOUNTADMIN'
+account = "<your account in form of organisation-account_name>"
+user = "<your user>"
+password = "<your password>"
+role = "<your role>"
+host="<host of your account, e.g. organisation-account_name.snowflakecomputing.com>"
 ```
 
-**Note: PRs for new resources will not be accepted without passing acceptance tests.**
+To be able to run all the tests you additionally need two more profiles `[secondary_test_account]` and `[incorrect_test_profile]`:
```

Comment:
Maybe it would be worth mentioning that some of the tests require the privileged role (preferably the`ACCOUNTADMIN`) and certain Snowflake Edition (preferably the highest available).

Reply:
Good point, I will add it.